inst/include/population_dynamics/fleet/fleet.hpp

```diff
@@ -112,13 +116,113 @@ struct Fleet : public FIMSObject<Type> {
   void Prepare() {
     // for(size_t fleet_ = 0; fleet_ <= this->nfleets; fleet_++) {
     // this -> Fmort[fleet_] = fims::exp(this -> log_Fmort[fleet_]);
+
+  // derived quantities
+  std::fill(catch_at_age.begin(), catch_at_age.end(), 0);    /*!<derived quantity catch at age*/
+  std::fill(catch_index.begin(), catch_index.end(), 0);     /*!<derived quantity catch index*/
+  std::fill(age_composition.begin(), age_composition.end(), 0);
+  std::fill(expected_catch.begin(), expected_catch.end(), 0); /*!<model expected total catch*/
+  std::fill(expected_index.begin(), expected_index.end(), 0); /*!<model expected index of abundance*/
+  std::fill(catch_numbers_at_age.begin(), catch_numbers_at_age.end(), 0); /*!<model expected catch at age*/
+  std::fill(catch_weight_at_age.begin(), catch_weight_at_age.end(), 0);  /*!<model expected weight at age*/
+      this->q = fims::exp(this->log_q);
     for (size_t year = 0; year < this->nyears; year++) {
+      FIMS_LOG << "input F mort " << this->log_Fmort[year] << std::endl;
+      FIMS_LOG << "input q " << this->log_q << std::endl;
       this->Fmort[year] = fims::exp(this->log_Fmort[year]);
-      this->q[year] = fims::exp(this->log_q[year]);
+
     }
   }
+  /**
+ * @brief Method to report out the fleet-specific derived quantities. 
+ * 
+ */
+  void ReportFleet(){
+    #ifdef TMB_MODEL
+      typename ModelTraits<Type>::EigenVector exp_index =
+        expected_index;
+      REPORT_F(exp_index, of);
+      typename ModelTraits<Type>::EigenVector exp_catch =
+        expected_catch;
+      REPORT_F(exp_catch, of);
+      typename ModelTraits<Type>::EigenVector F_mort =
+        Fmort;
+      REPORT_F(F_mort, of);
+    #endif
+}
+
+
+    virtual const Type evaluate_age_comp_ll() {
+      Type nll = 0.0; /*!< The negative log likelihood value */
+      #ifdef TMB_MODEL
+        fims::Dmultinom<Type> dmultinom;
+        size_t dims = 360;
```

Comment:
The value, 360, was hard coded to match the model comparison project and fix a bug.  We need to switch back to code as implemented in the [integration-test-R](https://github.com/NOAA-FIMS/FIMS/blob/88942e9da522d56aaa4eb86dcd9483d76b3a4806/inst/include/population_dynamics/fleet/fleet.hpp#L149) branch or develop a different generalized approach to assigning dims.